catalog/spatial/registry/registry-rest-endpoint/src/main/java/org/codice/ddf/registry/rest/endpoint/RegistryRestEndpoint.java

```diff
@@ -0,0 +1,171 @@
+ */
+
+package org.codice.ddf.registry.rest.endpoint;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.registry.federationadmin.service.FederationAdminException;
+import org.codice.ddf.registry.federationadmin.service.FederationAdminService;
+import org.codice.ddf.registry.federationadmin.service.RegistryPublicationService;
+import org.codice.ddf.registry.rest.endpoint.report.RegistryReportHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.jknack.handlebars.Handlebars;
+import com.github.jknack.handlebars.Template;
+import com.github.jknack.handlebars.io.ClassPathTemplateLoader;
+
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryPackageType;
+
+@Path("/")
+public class RegistryRestEndpoint {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RegistryRestEndpoint.class);
+
+    private FederationAdminService federationAdminService;
+
+    private ClassPathTemplateLoader templateLoader;
+
+    private RegistryPublicationService registryPublicationService;
+
+    private RegistryReportHelper reportHelper;
+
+    public RegistryRestEndpoint() {
+        templateLoader = new ClassPathTemplateLoader();
+        templateLoader.setPrefix("/templates");
+        templateLoader.setSuffix(".hbt");
+    }
+
+    @Path("/{registryId}/publication/{sourceId}")
+    @POST
+    public Response publish(@PathParam("registryId") String registryId,
+            @PathParam("sourceId") String destinationId) {
+        if (StringUtils.isBlank(registryId) || StringUtils.isBlank(destinationId)) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity("Registry and destination ID cannot be blank")
+                    .build();
+        }
+
+        try {
+            registryPublicationService.publish(registryId, destinationId);
+        } catch (FederationAdminException e) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity("Error publishing node to destination " + destinationId)
+                    .build();
+        }
+
+        return Response.status(Response.Status.OK)
+                .build();
+    }
+
+    @Path("/{registryId}/publication/{sourceId}")
+    @DELETE
+    public Response unpublish(@PathParam("registryId") String registryId,
+            @PathParam("sourceId") String destinationId) {
+        if (StringUtils.isBlank(registryId) || StringUtils.isBlank(destinationId)) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity("Registry and destination ID cannot be blank")
```

Comment:
Since you are doing the same parameter validation of registry ID and destination ID in each of these methods you should extract that logic out to reduce the code duplication.